### DISCUSSION TITLE
LINQ Methods and Dynamic Sorting

### DISCUSSION THREAD

**Discussion by JohnrFinney** [microsoft/PowerPlatform-DataverseServiceClient#268] [Q&A] [answered]
Hi,

Using the client, I can't figure a way to use Linq Methods to do dynamic sorting. The only way I know of is using Linq query syntax and I'm getting an error ....... "The method 'OrderBy' cannot follow the method 'Select' or is not supported. Try writing the query in terms of supported methods or call the 'AsEnumerable' or 'ToList' method before calling unsupported methods."

This is the following code and the orderByDynamic works against other data sources (Ef.core)..
        public List<Data.Contact> ListClaimantsByClassActionId2(Guid classActionId)
        {
            var res = (from e in this.Context.ContactSet
                       join cac in this.Context.cp_classactionclaimantSet
                          on e.Id equals cac.cp_claimantid.Id
                       where cac.cp_classaction.Id == classActionId
                        && cac.statecode == 0
                       select e).AsQueryable();

            res = QueryExtensions.OrderByDynamic<Data.Contact>(res, "cp_claimantstatus", false);

            return res.ToList();
        }

Is the problem above because there's a select in the Linq query?

My main issue is that of implementing dynamic sorting somehow..........  Does anyone have suggestions to implement dynamic sorting please?

**Comment by MattB-msft** [comment]
Is there any reason your cannot use the order by as part of the query? 
See: https://github.com/MicrosoftDocs/powerapps-docs/blob/main/powerapps-docs/developer/data-platform/org-service/linq-query-examples.md#use-the-orderby-operator

**Comment by JohnrFinney** [comment]
How would it be done to dynamically set the column/columns to be ordered within the Linq query?
________________________________

**Comment by MattB-msft** [comment, marked as answer]
In that scenario you would use late bound linq queries.. see here for an example [LateBoundQuery](https://github.com/microsoft/PowerApps-Samples/blob/d95e61673f2e6339054d9027730206af079b4f31/cds/orgsvc/C%23/QueriesUsingLINQ/LateBoundQuery/SampleProgram.cs#L62)

MattB
